conan/tools/google/bazeldeps.py

```diff
@@ -229,19 +235,90 @@ def load_conan_dependencies():
             )
         {% endfor %}
         """)
+    module_template = textwrap.dedent("""\
+        # This module provides a repo for each requires-dependency in your conanfile.
+        # It's generated by the BazelDeps, and should be used in your Module.bazel file.
+        load(":conan_deps_repo_rules.bzl", "conan_dependency_repo")
+
+        def _load_dependenies_impl(mctx):
+        {% for repository_name, pkg_folder, pkg_build_file_path in dependencies %}
+            conan_dependency_repo(
+                name = "{{repository_name}}",
+                package_path = "{{pkg_folder}}",
+                build_file_path = "{{pkg_build_file_path}}",
+            )
+        {% endfor %}
+
+            return mctx.extension_metadata(
+                root_module_direct_deps = None,
+                root_module_direct_dev_deps = None,
+
+                # Prevent writing function content to lockfiles:
+                # - https://bazel.build/rules/lib/builtins/module_ctx#extension_metadata
+                # Important for remote build. Actually it's not reproducible, as local paths will
+                # be different on different machines. But we assume that conan works correctly here.
+                # IMPORTANT: Not compatible with bazel < 7.1
+                reproducible = True,
```

Comment:
One oddity I noticed is that if I manually edit one of Conan's BUILD.bazel files after an initial build, the edited version does not get picked up, unless I delete MODULE.bazel.lock. If this isn't fixable without impacting performance, then maybe some documentation could be added to explain what to do.

Reply:
That's surprising because the repo rule below symlinks in the build file, so changes should be picked up no matter what Bazel does. Can you reproduce this with `--nosystem_rc --nohome_rc`?

Reply:
Ah, I see. That can be fixed like this:
https://github.com/bazelbuild/bazel/blob/fb171884e10787475b2936607e8c92717f7ff474/tools/build_defs/repo/local.bzl#L99

Reply:
Could the watch be done only if symlinks are not enabled?

Reply:
That's difficult to detect (would require checking the file type after the operation), but unnecessarily watching a few build files shouldn't matter that much.

Reply:
@Wyverald What do you think, should we `watch` automatically if the symlink is a copy?

Reply:
The automatic watch sounds like a great idea to me

Reply:
Can this be done in a follow up?

Reply:
Sure!